File: ShootR/ShootR/Server/Bullet/BulletMovementController.cs
using System;
using System.Diagnostics;

namespace ShootR
{
    public class BulletMovementController : MovementController
    {
        public const double MAX_SPEED = 1100;
        public const double MASS = 800;

        public BulletMovementController(Vector2 position, Vector2 direction, double rotation)
            : base(MASS, MAX_SPEED)
        {
            Position = position;
            Velocity = direction * Power;
            Rotation = rotation;
        }

        public override void Move(double percentOfSecond)
        {
            var incrementor = Velocity * percentOfSecond;

            Position += incrementor;
        }

        public override void Update(double PercentOfSecond)
        {
            Move(PercentOfSecond);
        }
    }
}

File: ShootR/ShootR/Client/Bullet/BulletMovementController.ts
/// <reference path="../Collidable/MovementControllers/MovementController.ts" />

class BulletMovementController extends MovementController {
    static MAX_SPEED: number = 1100; // Updated from server configuration value
    static MASS: number = 800;

    constructor () {
        super(BulletMovementController.MASS, BulletMovementController.MAX_SPEED);
    }

    public Move(percentOfSecond: number): void {
        var incrementor = Vector2.MultiplyN(this.Velocity, percentOfSecond);

        this.Position.AddV(incrementor);
    }

    public Update(percentOfSecond: number, now: Date): void {
        this.Move(percentOfSecond);
        super.Update(percentOfSecond, now);
    }
}